corehq/apps/locations/permissions.py

```diff
@@ -164,12 +164,24 @@ class LocationResource(HqBaseResource):
     return resource
 
 
+def conditionally_location_safe(conditional_function):
```

Comment:
@esoergel, not sure if you think this is a good reusable solution. Unfortunately, on class based views, we aren't given access to the instance of the view class that is getting called (I think because it hasn't been instantiated when we do the check), so there's some funky things that need to happen to do proper checks on the contents of the view.